Add Receiver concept

dennisgove opened the pull request:
Adds the concept of a Receiver which will be used to power the main application. This removes Transmitter from the main application code and instead allows us to wrap various types of Transmitters in Receiver objects. Doing this allows us to create test-only receivers and makes us less dependent on the implementation of various Transmitters.

Still work to be done to actually integrate this. And of course wrapping up xDripG5.Transmitter.

dennisgove committed:
Adds Receiver concept to wrap Transmitter.

dennisgove committed:
Adds static receiver for testing

dennisgove commented:
@loudnate, I think this is the most prudent approach for myself. It'll segregate my application code from any particular changes you make in the xDripG5 library.

loudnate commented on glucose-badge/ReceiverNotificationDelegate.swift:
```diff
@@ -0,0 +1,21 @@
+    func receiver(receiver: Receiver, didReceiveReading: Reading)
+
+    // TODO: decide if we need a designated error callback. Perhaps Reading can
+    // become a protocol that can have GlucoseReading and ErrorReading
```
Maybe consider returning an enum like `Either<GlucoseReading, ErrorType>`

dennisgove committed:
Integreated with application

Now creates static receiver which will write the same 2 values to the badge every 2.5 seconds.
